src/main/java/com/minecolonies/coremod/colony/buildings/BuildingBarracks.java

```diff
@@ -109,6 +109,30 @@ public void onUpgradeComplete(final int newLevel)
                     ((BuildingBarracksTower) building).addBarracks(getLocation());
                 }
             }
+
+            //TODO: Implement the following properly. Once we've got textures and etc for a Substitution block.
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1135)